hooks/src/index.js

```diff
@@ -346,7 +350,11 @@ function invokeCleanup(hook) {
 	// A hook cleanup can introduce a call to render which creates a new root, this will call options.vnode
 	// and move the currentComponent away.
 	const comp = currentComponent;
-	if (typeof hook._cleanup == 'function') hook._cleanup();
+	let cleanup;
+	if (typeof (cleanup = hook._cleanup) == 'function') {
```

Comment:
should compress to the same code:

```suggestion
	let cleanup = hook._cleanup;
	if (typeof cleanup == 'function') {
```